app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -95,7 +95,11 @@ def destroyable_associations
   end
 
   def destroyable_association?(reflection)
-    %i[destroy delete_all].include?(reflection.options[:dependent])
+    if object.respond_to?(:background_destroy_associations)
+      object.background_destroy_associations.include?(reflection.name)
+    else
+      raise "The old way! BackgroundDeletation should be used for #{object.class.name}"
```

Comment:
Can you use a tagged error and correct the "deletation" to "deletion"?

Reply:
this won't be here.. it's just for CI.

Reply:
but when I thinking about it, maybe it would make it.. so will change it.